### PR TITLE
refactor secret controller

### PR DESCRIPTION
**Pull request by sbezverk**
Refactor secret controller, now the controller watches for all secrets with a specific label. This approach allows to track if the secret was created but the label was added/removed later and no extra logic requires for updates This PR also adds removal of a remote cluster from the slice of clusters.

### CODE REVIEW

**Comment by gyliu513** [comment]
Fixed https://github.com/istio/istio/issues/5440

**Comment by gyliu513** [comment]
Thanks @sbezverk , also test in my local cluster and it works fine.

```
2018-05-08T07:05:31.476744Z	info	Processing delete: istio-system/myistio.icp
2018-05-08T07:05:31.476783Z	info	Deleting cluster member: myistio.icp
2018-05-08T07:05:31.476789Z	info	Number of clusters in the cluster store: 0
2018-05-08T07:06:05.964373Z	info	Processing add: istio-system/myistio.icp
2018-05-08T07:06:05.964410Z	info	Adding new cluster member: myistio.icp
2018-05-08T07:06:05.965273Z	info	Number of clusters in the cluster store: 1
```

FYI @sdake

**Comment by sdake** [comment]
Will this work with the existing code to populate the secrets?  You might consider removing that code in this PR.

**Comment by sbezverk** [comment]
@sdake will do.

File: pilot/pkg/bootstrap/server.go [hunk 1]
```diff
@@ -275,9 +275,6 @@ func (s *Server) initClusterRegistries(args *PilotArgs) (err error) {
 			s.clusterStore); err != nil {
 			return err
 		}
-	} else if err = clusterregistry.ReadClustersV2(s.kubeClient,
```
Comment (gyliu513) [inline pilot/pkg/bootstrap/server.go]: @sbezverk @sdake  if this code is removed, then how to handle the case that the secret was created prior to create the istio control panel?

Comment (gyliu513) [reply pilot/pkg/bootstrap/server.go]: Ah, I see, once the pilot is start, it will detect the secret and add the cluster.

Comment (sbezverk) [reply pilot/pkg/bootstrap/server.go]: @gyliu513 exactly, controller will get notification for every already existing secrets matching List/Watch criteria.

**Comment by sbezverk** [comment]
@costinm Please review, @gyliu513  has tested it and it worked in his multi cluster env.

**Pull request merged**